File: Modules/TaskFlow/Tag/TagColumns.cs
namespace TaskFlowSerenity.TaskFlow.Columns;

[ColumnsScript("TaskFlow.Tag")]
[BasedOnRow(typeof(TagRow), CheckNames = true)]
public class TagColumns
{
    [EditLink]
    public string Name { get; set; }
    public string Color { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Modules/ServerTypes/TaskFlow/TagColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TagRow } from "./TagRow";

export interface TagColumns {
    Name: Column<TagRow>;
    Color: Column<TagRow>;
    CreatedAt: Column<TagRow>;
    UpdatedAt: Column<TagRow>;
}

export class TagColumns extends ColumnsBase<TagRow> {
    static readonly columnsKey = 'TaskFlow.Tag';
    static readonly Fields = fieldsProxy<TagColumns>();
}